src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -1,93 +1,604 @@
+    public async ValueTask<QuicStream> OpenOutboundStreamAsync(QuicStreamType type, CancellationToken cancellationToken = default)
+    {
+        ObjectDisposedException.ThrowIf(_disposed == 1, this);
 
-        public EndPoint RemoteEndPoint => _provider.RemoteEndPoint;
+        QuicStream stream = new QuicStream(new Implementations.MsQuic.MsQuicStream(_state, _handle, type));
+        try
+        {
+            await stream.StartAsync(cancellationToken).ConfigureAwait(false);
+        }
+        catch
+        {
+            await stream.DisposeAsync().ConfigureAwait(false);
+            throw;
+        }
+        return stream;
+    }
 
-        public X509Certificate? RemoteCertificate => _provider.RemoteCertificate;
+    /// <summary>
+    /// Accepts an inbound <see cref="QuicStream" />.
+    /// </summary>
+    /// <param name="cancellationToken">A cancellation token that can be used to cancel the asynchronous operation.</param>
+    /// <returns>An asynchronous task that completes with the accepted <see cref="QuicStream" />.</returns>
+    public async ValueTask<QuicStream> AcceptInboundStreamAsync(CancellationToken cancellationToken = default)
+    {
+        ObjectDisposedException.ThrowIf(_disposed == 1, this);
 
-        public SslApplicationProtocol NegotiatedApplicationProtocol => _provider.NegotiatedApplicationProtocol;
+        if (!_canAccept)
+        {
+            throw new InvalidOperationException(SR.net_quic_accept_not_allowed);
+        }
 
-        internal ValueTask FinishHandshakeAsync(QuicServerConnectionOptions options, string? targetHost, CancellationToken cancellationToken = default) => _provider.FinishHandshakeAsync(options, targetHost, cancellationToken);
+        try
+        {
+            return await _acceptQueue.Reader.ReadAsync(cancellationToken).ConfigureAwait(false);
+        }
+        catch (ChannelClosedException ex) when (ex.InnerException is not null)
+        {
+            ExceptionDispatchInfo.Capture(ex.InnerException).Throw();
+            throw;
+        }
+    }
 
-        /// <summary>
-        /// Connect to the remote endpoint.
-        /// </summary>
-        /// <param name="cancellationToken"></param>
-        /// <returns></returns>
-        public ValueTask ConnectAsync(CancellationToken cancellationToken = default) => _provider.ConnectAsync(cancellationToken);
+    /// <summary>
+    /// Closes the connection with the application provided code, see <see href="https://www.rfc-editor.org/rfc/rfc9000.html#immediate-close">RFC 9000: Connection Termination</see> for more details.
+    /// </summary>
+    /// <remarks>
+    /// Connection close is not graceful in regards to its streams, i.e.: calling <see cref="CloseAsync(long, CancellationToken)"/> will immediately abort all streams associated with this connection.
+    /// Please make sure, that all streams have been closed and all their data consumed before calling this method;
```

Comment:
I don't think we should use the word "please" in comments. But I think we will need our docs team to proofread the comments anyway...

Reply:
Done